sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleHistogramAggregator.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.aggregator;
+
+import io.opentelemetry.api.common.Labels;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.metrics.common.InstrumentDescriptor;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.DoubleHistogramData;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
+import javax.annotation.concurrent.GuardedBy;
+
+final class DoubleHistogramAggregator extends AbstractAggregator<HistogramAccumulation> {
+  private final double[] boundaries;
+
+  DoubleHistogramAggregator(
+      Resource resource,
+      InstrumentationLibraryInfo instrumentationLibraryInfo,
+      InstrumentDescriptor instrumentDescriptor,
+      double[] boundaries,
+      boolean stateful) {
+    super(resource, instrumentationLibraryInfo, instrumentDescriptor, stateful);
+    this.boundaries = boundaries;
+  }
+
+  @Override
+  public AggregatorHandle<HistogramAccumulation> createHandle() {
+    return new Handle(this.boundaries);
+  }
+
+  @Override
+  public final HistogramAccumulation merge(HistogramAccumulation x, HistogramAccumulation y) {
+    if (!x.getBoundaries().equals(y.getBoundaries())) {
+      throw new IllegalArgumentException("can't merge histograms with different boundaries");
+    }
+
+    long[] mergedCounts = new long[x.getCounts().size()];
+    for (int i = 0; i < x.getCounts().size(); ++i) {
+      mergedCounts[i] = x.getCounts().get(i) + y.getCounts().get(i);
+    }
+    return HistogramAccumulation.create(
+        x.getCount() + y.getCount(),
+        x.getSum() + y.getSum(),
+        x.getBoundaries(),
+        Arrays.stream(mergedCounts).boxed().collect(Collectors.toList()));
```

Comment:
yeah, not sure why we can't just store these as arrays in the accumulation, rather than boxing them all up.

Reply:
iirc the reason is that AutoValue would complain about creating HistogramAccumulation with mutable primitive arrays.
but yep I could change it to the array approach and document the potential mutable issue.

Reply:
> We may as well construct the list up above instead of copying through a stream I think (by presizing and using set, not add).

yep, benchmark shows that it's much faster than boxing a stream so I've updated the codes accordingly.
```
Benchmark                    Mode  Cnt      Score     Error   Units
ArrayCopy.copy_arrayListLoop thrpt   10  12776.515 ± 437.142  ops/ms
ArrayCopy.copy_streamBoxed  thrpt   10   4524.458 ±  90.628  ops/ms
```

I'm not comfortable with primitive arrays since it's impossible to mark them as immutable. and I've wrapper the list as a unmodifiable one. `ImmutableList` would be ideal here but guava is not in the dependencies so out of options.

Reply:
a wrapper class around the array would probably be a viable option.

Reply:
> a wrapper class around the array would probably be a viable option.

100% agree. but it seems to me it would be reinventing the wheels like `ImmutableList`. and where should we put the class in the hierarchy? not sure sdk is the right place tho.

Reply:
ImmutableList boxes every item in the list individually, meaning you have n extra objects per list, rather than 1 extra object in a single wrapper for an array.

Reply:
are you suggesting something like [ImmutableLongArray](https://guava.dev/releases/22.0/api/docs/com/google/common/primitives/ImmutableLongArray.html) and [ImmutableDoubleArray](https://guava.dev/releases/22.0/api/docs/com/google/common/primitives/ImmutableDoubleArray.html) to completely get rid of the boxing? since #2148 I can't use these classes directly but have to implement a similar & simpler version in the sdk.  I don't have a strong opinion of that since I don't have much understanding of the project context, and I would add them if you think that's a good idea.

Reply:
Something like that, yes, but I don't think we need all the methods and functionality that the guava classes currently provide.

Reply:
I've replaced all read-only primitive arrays with ImmutableLongArray/ImmutableDoubleArray.